src/coreclr/inc/gcinfotypes.h

```diff
@@ -43,14 +47,28 @@ __forceinline size_t SAFE_SHIFT_RIGHT(size_t x, size_t count)
 
 inline UINT32 CeilOfLog2(size_t x)
 {
+    // we want lzcnt, but bsr is ok too
     _ASSERTE(x > 0);
-    UINT32 result = (x & (x - 1)) ? 1 : 0;
-    while (x != 1)
-    {
-        result++;
-        x >>= 1;
-    }
-    return result;
+
+    x = (x << 1) - 1;
+
+#ifdef TARGET_64BIT
+#ifdef _MSC_VER
+    DWORD lzcountCeil;
+    _BitScanReverse64(&lzcountCeil, (unsigned long)x);
+#else // _MSC_VER
+    UINT32 lzcountCeil = (UINT32)__builtin_clzl((unsigned long)x);
+#endif // _MSC_VER
+#else // TARGET_64BIT
+#ifdef _MSC_VER
+    DWORD lzcountCeil;
+    _BitScanReverse(&lzcountCeil, (unsigned long)x);
+#else // _MSC_VER
+    UINT32 lzcountCeil = (UINT32)__builtin_clz((unsigned int)x);
```

Comment:
I believe __builtin_clz already encodes the subtract from BITS_PER_SIZE_T within the intrinsic function unlike _BitScanReverse

Reply:
right, I've just realized that even though lzcnt is encoded similarly to bsr on x64, the result is different.